### PR TITLE
added k8s packages and dependencies

### PR DESCRIPTION
**Pull request by wh330**

### CODE REVIEW

File: packages/gcompat/APKBUILD
Comment (herry13) [inline packages/gcompat/APKBUILD]: Could you become the maintainer of this package?

File: packages/k8s-cni/APKBUILD
Comment (herry13) [inline packages/k8s-cni/APKBUILD]: Could you become the maintainer of this package?

File: packages/k8s-cni/APKBUILD
Comment (herry13) [inline packages/k8s-cni/APKBUILD]: Does this package only for `armhf` architecture?

File: packages/gcompat/APKBUILD
Comment (wh330) [reply packages/gcompat/APKBUILD]: Yes, sure!

File: packages/k8s/APKBUILD
Comment (herry13) [inline packages/k8s/APKBUILD]: Could you please become the maintainer of this package?

File: packages/k8s-cni/APKBUILD
Comment (wh330) [reply packages/k8s-cni/APKBUILD]: Yes, sure!

File: packages/k8s/APKBUILD
Comment (wh330) [reply packages/k8s/APKBUILD]: Yes, sure!

File: packages/k8s-cni/APKBUILD
Comment (wh330) [reply packages/k8s-cni/APKBUILD]: Yes, it's only for armhf.

File: packages/k8s/APKBUILD
Comment (herry13) [inline packages/k8s/APKBUILD:34]: Do you need to include the `kubelet` startup script in a particular runlevel e.g. `default`?

File: packages/k8s/kubelet.initd
Comment (herry13) [inline packages/k8s/kubelet.initd]: Could we remove line 13?

**Review by herry13 — CHANGES_REQUESTED** [review]
Hi @wajdihajji. I have a request of changes for this PR before merging it into master.

**Comment by herry13** [comment]
Many thanks @wajdihajji for this PR.
Is the `gcompat` package similar to the one available in Alpine's edge/community repo: https://pkgs.alpinelinux.org/packages?name=gcompat&branch=edge?

Hi @herry13 it's the same package. The GitHub repository I used (https://github.com/AdelieLinux/gcompat) is referring to the one used for the package available in Alpine's repo.

**Comment by herry13** [comment]
@wajdihajji I got the following error message when building `k8s` package:

```
>>> ERROR: k8s*: ld-linux-armhf.so.3: path not found
```

Do you know what's the caused of this?

@herry13 I've faced that issue and I'm still trying to figure out what is the reason behind. The good thing is that it has no effect on the packaging at the moment.

File: packages/k8s-cni/APKBUILD
Comment (herry13) [reply packages/k8s-cni/APKBUILD]: Then line 8 should be `arch="armhf"`

File: packages/gcompat/APKBUILD
Comment (herry13) [reply packages/gcompat/APKBUILD]: Please could you put your name/email at line 2?

File: packages/k8s-cni/APKBUILD
Comment (herry13) [reply packages/k8s-cni/APKBUILD]: Please could you put your name/email at line 2?

File: packages/k8s/APKBUILD
Comment (herry13) [reply packages/k8s/APKBUILD]: Please could you put your name/email at line 2?

File: packages/k8s/APKBUILD
Comment (wh330) [reply packages/k8s/APKBUILD:34]: Kubelet should be started manually when initialising master or adding a worker. I think it shouldn't be started on boot as some nodes might not be either a master or a worker. We'll figure out, later on, how to automatically add workers.

**Review by wh330 — COMMENTED** [review]
I've commented on your feedback

**Comment by herry13** [comment]
@wajdihajji please could you change the installation dir from `/data` to `/usr`?

I just uploaded FruitOS version 0.2.4 which has larger root-partitions (512MB). You can use it for testing. Please let me know if you need larger free space. Thanks.

You might need to `rebase` since branch master has changed.

**Comment by wh330** [comment]
Hi @herry13 I changed the path in the APKBUILD files from /data to /usr and I `rebase` `master` onto the `k8s` branch.

**Comment by herry13** [comment]
Thank you @wajdihajji. I fixed few incorrect paths in your scripts and merged the packages into branch master. They're now available on fruit repo.

**Pull request closed without merging**
